v2.1/online-schema-changes.md

```diff
@@ -0,0 +1,205 @@
+---
+title: Online Schema Changes
+summary: Update table schemas without external tools or downtime.
+toc: true
+---
+
+CockroachDB's online schema changes provide a simple way to update a table schema without imposing any negative consequences on an application — including downtime. The schema change engine is a built-in feature requiring no additional tools, resources, or ad hoc sequencing of operations.
+
+Benefits of online schema changes include:
+
+- Changes to your table schema happen while the database is running.
+- The schema change runs as a [background job][show-jobs] without holding locks on the underlying table data.
+- Your application's queries can run normally, with no effect on read/write latency. The schema is cached for performance.
+- Your data is kept in a safe, [consistent][consistent] state throughout the entire schema change process.
+
+{{site.data.alerts.callout_success}}
+Support for schema changes within transactions is limited. We recommend doing schema changes outside transactions where possible.  When a schema management tool uses transactions on your behalf, we recommend only doing one schema change operation per transaction.
+{{site.data.alerts.end}}
+
+{{site.data.alerts.callout_info}}
+Schema changes keep your data consistent at all times, but they do not run inside user transactions in the general case. This is necessary so the cluster can remain online and continue to service application reads and writes. For more information, see [Limitations](#limitations).
+{{site.data.alerts.end}}
+
+<span class="version-tag">New in v2.1:</span> You can run schema changes inside the same transaction as a [`CREATE TABLE`][create-table] statement. For more information, [see this example](#run-schema-changes-inside-a-transaction-with-create-table).
+
+## How online schema changes work
+
+At a high level, online schema changes are accomplished by using a bridging strategy involving concurrent uses of multiple versions of the schema. Schema changes are run as a series of transactions:
+
+1. A user-level transaction initiated by [`ALTER TABLE`][alter-table], [`CREATE INDEX`][create-index], [`TRUNCATE`][truncate], etc. The user-level transaction schedules the schema change for the schema change engine to complete.
```

Comment:
I think we may be overloading the term "transaction" in this doc. Given the callouts at the top and the limitations section about schema changes and transactions, we probably want to use different terminology here, for example:

```
... The process is as follows:

1. A user initiates a schema change by executing `ALTER TABLE`, `CREATE INDEX`, ...
2. The schema change engine converts the original schema to...
```

Thoughts?

Reply:
I agree!  (Hopefully) fixed by updating using your proposed language.  More broadly, all mention of "system" transactions has been removed, and we just note that schema changes do what they do (moving your data, protecting it, loving it) without quite going into the how (which exposes the potentially confusing system txns concept).